### PR TITLE
wordMVC

### PR DESCRIPTION
**Pull request by nguyenquangminh**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,7 +3,8 @@ gem "rails",          "4.2.4"
 gem "config",         "1.2.1"
 gem "bcrypt",         ">= 3.1.11"
 gem "mysql2",         "~> 0.3.18"
-gem "bootstrap-sass", "~> 3.3.6" 
+gem "will_paginate", "3.1.0"
+gem "bootstrap-sass", "~> 3.3.6"
```
Comment (trantube94) [inline Gemfile]: xóa đi thêm lại?

File: app/assets/javascripts/words.coffee [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
+# All this logic will automatically be available in application.js.
```
Comment (trantube94) [inline app/assets/javascripts/words.coffee]: xóa

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -3,4 +3,10 @@ class ApplicationController < ActionController::Base
   # For APIs, you may want to use :null_session instead.
   protect_from_forgery with: :exception
   include SessionsHelper
+
+  def is_admin?
+    if current_user.admin != 1 redirect_to root_path
```
Comment (trantube94) [inline app/controllers/application_controller.rb]: enum

File: app/views/words/show.html.erb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+        <%= @word.category_id %>
+        <% end %><br>
+      </li>
+
+
+    </ul>
+  <% end %>
+
+  <% if current_user.is_admin? %>
+  <%= link_to t("admin.word.delete"), {controller: "words", action: "destroy", id: @word}, method: :delete,
```
Comment (trantube94) [inline app/views/words/show.html.erb]: rake routes

File: app/controllers/application_controller.rb (hunk 3 above)
Comment (quyettammanh) [inline app/controllers/application_controller.rb]: def is_admin?
  redirect_to(root_path) unless current_user.admin?

File: app/controllers/words_controller.rb [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    else
+      flash.now[:danger] = t("words.status.fail")
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
```
Comment (quyettammanh) [inline app/controllers/words_controller.rb]: def update
  if is_admin1?
    if @word.update_attributes(word_params)
       flash[:success] = t("words.status.success")
       redirect_to @word
     else
       flash.now[:danger] = t("words.status.fail")
       render :edit
     end
  else 
     redirect_to(root_path)
  end
end

File: Gemfile [hunk 6]
```diff
@@ -3,14 +3,15 @@ gem "rails",          "4.2.4"
...
-gem "bootstrap-sass", "~> 3.3.6" 
+gem "bootstrap-sass", "~> 3.3.6"
 gem "sass-rails",     "~> 5.0"
 gem "uglifier",       ">= 1.3.0"
 gem "coffee-rails",   "~> 4.1.0"
 gem "jquery-rails"
 gem "turbolinks"
 gem "jbuilder",       "~> 2.0"
 gem "sdoc",           "~> 0.4.0", group: :doc
+gem "will_paginate", "3.1.0"
```
Comment (quyettammanh) [inline Gemfile]: ko thang hang

File: app/helpers/words_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module WordsHelper
+end
```
Comment (minhth1905) [inline app/helpers/words_helper.rb]: file ko dung thì xóa đi em

File: app/views/words/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+</div>
+
+<ul class="wordlist">
+  <% if @words.empty? %>
+      <h2><%= t("word.notification") %></h2>
+  <% else %>
+    <% if params[:choice] %>
+      <% @words.each do |word| %>
+        <li class="wordlist-element">
+            <p><%= word.name %></p>
```
Comment (minhth1905) [inline app/views/words/index.html.erb]: indent

File: app/views/words/index.html.erb [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      <%= label_tag t("word.label.all") %><br>
+      <%= submit_tag t("word.filter") %>
+      <%= button_to "PDF" %><br>
+    <% end %>
+  </div>
+</div>
+
+<ul class="wordlist">
+  <% if @words.empty? %>
+      <h2><%= t("word.notification") %></h2>
```
Comment (minhth1905) [inline app/views/words/index.html.erb]: indent

File: app/views/words/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<%= provide(:title, t("word.new"))%>
```
Comment (minhth1905) [inline app/views/words/new.html.erb]: space

File: config/locales/en.yml [hunk 11]
```diff
@@ -29,27 +29,27 @@ en:
     subject: "SUBJECT"
     page: "PAGE"
     content: "This is class online. Page is used to learn Subject such as Math, English,
```
Comment (minhth1905) [inline config/locales/en.yml]: sap xep lại

File: app/models/word.rb [hunk 12]
```diff
@@ -1,7 +1,27 @@
 class Word < ActiveRecord::Base
+
+  AUDIO_URL = 'https://ssl.gstatic.com/dictionary/static/sounds/de/0/'
```
Comment (minhth1905) [inline app/models/word.rb]: ' => "

File: app/views/words/_form.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+  <%= f.text_field :name, class: "form-control" %><br>
+  <%= f.collection_select(:category_id, Category.all, :id, :title)%>
+
+  <%= f.fields_for :answers do |a| %>
+    <%= a.label :answer %>
+    <%= a.text_field :answer, class: "form-control" %>
+    <%= a.check_box :status, class: "form-control" %><br>
+  <% end %>
+
+  <%= f.submit t("word.submit") class: "btn btn-primary"%>
```
Comment (minhth1905) [inline app/views/words/_form.html.erb]: space

File: app/views/words/_form.html.erb [hunk 14]
```diff
@@ -0,0 +1,14 @@
+<%= form_for @word do |f| %>
+  <%= f.label :name %>
+  <%= f.text_field :name, class: "form-control" %><br>
+  <%= f.collection_select(:category_id, Category.all, :id, :title)%>
```
Comment (minhth1905) [inline app/views/words/_form.html.erb]: space

File: app/views/words/index.html.erb (hunk 9 above)
Comment (minhth1905) [inline app/views/words/index.html.erb]: indent

File: app/controllers/words_controller.rb [hunk 15]
```diff
@@ -0,0 +1,72 @@
...
+    else
+      flash.now[:danger] = t("words.status.fail_delete")
+    end
+    redirect_to words_url
+  end
+
+  private
+  def word_params
+    params.require(:word).permit :name, :category_id, :audio,
+      answers_attributes: [:id, :answer, :status ,:_destroy]
```
Comment (minhth1905) [inline app/controllers/words_controller.rb]: space

File: app/controllers/words_controller.rb [hunk 16]
```diff
@@ -0,0 +1,72 @@
...
+        @words = Word.allword(id: params[:category_id])
+      elsif params[:choice] == "Learned"
+        @words = Word.where_current_user_id(id: current_user.id).learned
+      elsif params[:choice] == "Unlearned"
+        @words = Word.where_current_user_id(id: current_user.id).unlearned
+      else
+        @words = Word.alphabet(id: params[:category_id])
+      end
+    end
+    @words = Word.paginate(:page => params[:page], :per_page => 30)
```
Comment (trantube94) [inline app/controllers/words_controller.rb]: ko dung =>, Settings

File: app/views/words/show.html.erb [hunk 17]
```diff
@@ -0,0 +1,42 @@
+<%= provide(:title, @word.name) %>
+
+<h1><%= @word.name %></h1>
+
+  <% if @word.empty? %>
+    <%= t("word.empty") %>
+  <% else %>
+    <ul class="word">
+      <li class="word-element">
+        <p>Name : </p>
```
Comment (trantube94) [inline app/views/words/show.html.erb]: I18n

**Pull request closed without merging**